query/src/main/kotlin/com/linecorp/kotlinjdsl/query/clause/from/Joiner.kt

```diff
@@ -1,12 +1,14 @@
+@file:Suppress("UNCHECKED_CAST")
+
 package com.linecorp.kotlinjdsl.query.clause.from
 
 import com.linecorp.kotlinjdsl.query.spec.*
 import com.linecorp.kotlinjdsl.query.spec.expression.EntitySpec
 import java.util.*
-import javax.persistence.criteria.AbstractQuery
-import javax.persistence.criteria.From
-import javax.persistence.criteria.JoinType
-import javax.persistence.criteria.Root
+import javax.persistence.criteria.*
+
+internal typealias ExplicitErasedParent = Number
+internal typealias ExplicitErasedChild = Int
```

Comment:
-- korean
어떤 용도로 사용되는 것인가요? 

-- english
What is it used for?

Reply:
Entities coming into Treat must be explicit parent/child relationship entities. However, we accept all cases and have no special restrictions. However, when using the actual treat , a parent/child relationship is clearly needed, and for this purpose, the type erasing part in the generic was used to trick the compiler.
-- korean
Treat 으로 들어오는 엔티티는 명시적인 부모/자식 관계 엔티티여야 합니다. 그러나 우리는 모든 클래스의 경우를 다 받고 있고 특별한 제약을 두고 있지 않습니다. 그러나 실제 treat 을 사용할 때에는 명확히 부모/자식 관계가 필요하며 이를 위해 generic 에서 type 이 erasing 되는 부분을 이용해 컴파일러를 속이는 작업을 수행했습니다.

Reply:
So, I did a forced casting to this generic type. The clear parent-child relationship we know is Number (parent) and Int (child)

-- korean
그래서 이 generic type으로 강제 casting 하는 작업을 수행했습니다. 우리가 아는 명확한 부모 자식 관계는 Number(부모)와 Int(자식)가 있습니다

Reply:
-- korean
그러면 그 속이는 부분은 Joiner 안에서만 수행되면 될 것으로 보입니다. private로 변경하는 것은 어떤가요?
추가로 이러한 내용이 주석으로 있으면 좋을 것 같습니다. 왜냐하면 Number와 Int 타입으로는 이러한 배경을 유추하기 힘들기 때문입니다.

-- english
Then, it seems that the trick logic only is needed only within the Joiner. How about changing to private?
Additionally, it would be good to have comments about these background. Because it is hard to know this background with Number and Int type.

Reply:
This is also used in tests, so I limited it to internal as much as possible.

-- korean
이게 테스트에서도 사용하고 있어서 internal 로 최대한 제약했습니다.

Reply:
After thinking about what you said, I can change it, so I created an alias for the test and changed it to private.

-- korean
말씀하신 내용을 고민해보니 바꿀 수 있어서 테스트도 별도로 alias 를 만들고 private 으로 전환했습니다.